react/features/calendar-sync/actions.js

```diff
@@ -242,3 +243,42 @@ export function updateProfile(calendarType: string): Function {
             });
     };
 }
+
+/**
+ * Updates calendar event by generating new invite URL and editing the event
+ * adding some descriptive text and location.
+ *
+ * @param {string} id - The event id.
+ * @param {string} calendarId - The id of the calendar to use.
+ * @returns {Function}
+ */
+export function updateCalendarEvent(id: string, calendarId: string): Function {
+    return (dispatch: Dispatch<*>, getState: Function) => {
+
+        const { integrationType } = getState()['features/calendar-sync'];
+        const integration = _getCalendarIntegration(integrationType);
+
+        if (!integration) {
+            return Promise.reject('No integration found');
+        }
+
+        const { locationURL } = getState()['features/base/connection'];
+        const newRoomName = generateRoomWithoutSeparator();
+        let href = locationURL.href;
+
+        href.endsWith('/') || (href += '/');
+
+        const roomURL = `${href}${newRoomName}`;
+
+        return dispatch(integration.updateCalendarEvent(
+                id, calendarId, roomURL))
+            .then(() => {
+                const { events } = getState()['features/calendar-sync'];
+                const eventIx = events.findIndex(
+                    e => e.id === id && e.calendarId === calendarId);
+
+                events[eventIx].url = roomURL;
```

Comment:
I'm not sure how I feel about modifying the event in place because philosophically the state in redux accessed directly from outside reducers should be read-only; any change should be done through actions.

Reply:
One line below is the dispatch to update the redux ... so we need to clone the whole list of events and then do the modification, this is the right approach right?

Reply:
That or expose another action so that the update happens in the reducer where the cloning would happen anyway, or maybe clone the individual event and have redux replace the old event with the new one. But yeah, some kind fo cloning.